ts/src/main/scala/ch/epfl/ts/engine/MarketRules.scala

```diff
@@ -47,13 +53,13 @@ class MarketRules extends Serializable {
                        matchExists: (Double, Double) => Boolean = alwaysTrue,
                        oldTradingPrice: Double,
                        enqueueOrElse: (Order, PartialOrderBook) => Unit): Double = {
```

Comment:
Could this function be refactored? Right now it's very difficult to read and understand what it does.